cmd/sdn-cni-plugin/openshift-sdn_linux.go

```diff
@@ -94,25 +93,33 @@ func (p *cniPlugin) doCNI(url string, req *cniserver.CNIRequest) ([]byte, error)
 
 // Send the ADD command environment and config to the CNI server, returning
 // the IPAM result to the caller
-func (p *cniPlugin) doCNIServerAdd(req *cniserver.CNIRequest, hostVeth string) (types.Result, error) {
+func (p *cniPlugin) doCNIServerAdd(req *cniserver.CNIRequest, hostVeth string) (*current.Result, error) {
 	req.HostVeth = hostVeth
 	body, err := p.doCNI("http://dummy/", req)
 	if err != nil {
 		return nil, err
 	}
 
-	// We currently expect CNI version 0.2.0 results, because that's the
+	// We currently expect CNI version 0.3.1 results, because that's the
 	// CNIVersion we pass in our config JSON
-	result, err := types020.NewResult(body)
+	result, err := current.NewResult(body)
 	if err != nil {
 		return nil, fmt.Errorf("failed to unmarshal response '%s': %v", string(body), err)
 	}
 
-	return result, nil
+	return result.(*current.Result), nil
 }
 
 func (p *cniPlugin) testCmdAdd(args *skel.CmdArgs) (types.Result, error) {
-	return p.doCNIServerAdd(newCNIRequest(args), "dummy0")
+	result, err := p.doCNIServerAdd(newCNIRequest(args), "dummy0")
+	if err != nil {
+		return nil, err
+	}
+	convertedResult, err := convertToRequestedVersion(args.StdinData, result)
```

Comment:
Just do `return convertToRequestedVersion(args.StdinData, result)` here; no need for the other 4 lines if we're not doing any additional error processing or touching convertedResult.

Reply:
Fixed. Thanks.